distributed/diagnostics/plugin.py

```diff
@@ -175,6 +176,40 @@ def release_key(self, key, state, cause, reason, report):
         """
 
 
+class NannyPlugin:
+    """Interface to extend the Nanny
+
+    A worker plugin enables custom code to run at different stages of the Workers'
+    lifecycle. A nanny plugin does the same thing, but benefits from being able
+    to run code before the worker is started, or to restart the worker if
+    necessary.
+
+    To implement a plugin implement some of the methods of this class and register
+    the plugin to your client in order to have it attached to every existing and
+    future workers with ``Client.register_nanny_plugin``.
```

Comment:
I don't think ``Client.register_nanny_plugin`` exists. It looks like we're going with a `nanny=True` keyword to `Client.register_worker_plugin`` instead